saidump/saidump.cpp

```diff
@@ -399,7 +418,159 @@ void dumpGraph(const TableDump& td)
     std::cout << "}" << std::endl;
 }
 
-int main(int argc, char ** argv)
+/*
+ preprocess the input json file to make sure it's a valid json file.
+*/
+int pre_process_file(string file_name)
+{
+    ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        cerr << "Failed to open the input file." << endl;
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.\n", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+    input_file.seekg(0, ios::beg);     // Move to the begin of the file
+
+    // Read the content of the input file into a string
+    string content((istreambuf_iterator<char>(input_file)),
+                   istreambuf_iterator<char>());
+
+
+    input_file.close();
+
+    content = regex_replace(content,  regex("\\},\\{\\r"), ",");
+
+    //erase the 1st and last char.
+    if (content.length() >= 2 && content[0] == '[' && content[content.length()-1] == ']')  {
+        content.erase(0, 1);
+        content.erase(content.length() - 1);
+    }
+
+    ofstream outputFile(file_name);
+
+    if (!outputFile.is_open())
+    {
+        cerr << "Failed to open the output file." << endl;
+        return SAI_STATUS_FAILURE;
+    }
+
+    outputFile << content;
+    outputFile.close();
+
+    return SAI_STATUS_SUCCESS;
+}
+
+int dump_from_redis_rdb_file(string file_name)
+{
+    ifstream input_file(file_name);
+    if (!input_file.is_open())
+    {
+        SWSS_LOG_NOTICE("The file %s does not exist for dumping from redis dump.rdb file.", file_name);
```

Comment:
also not printing to stderro, maybe you need new log SWSS_LOG_STD_ERROR etc to print in both places ?

Reply:
Fix it.